static/gulpfile.js

```diff
@@ -0,0 +1,104 @@
+/* @file gulpfile.js
+ * @brief Builds front-end static files.
+ *
+ * @author Oscar Bezi (oscar@bezi.io)
+ */
+'use strict';
+
+//==============================================================================
+// Includes
+//==============================================================================
+var gulp = require('gulp');
+var util = require('gulp-util');
+
+var watchify = require('watchify');
+var browserify = require('browserify');
+
+var source = require('vinyl-source-stream');
+var buffer = require('vinyl-buffer');
+
+var assign = require('lodash.assign');
+
+var uglify = require('gulp-uglify');
+
+//==============================================================================
+// Config
+//==============================================================================
+var srcDir = './src/'
+var destDir = './build/'
+var bundleFiles = ['admin.js', 'index.js'];
+
+/* @brief Makes a JS build task for the given bundle files.
+ *
+ * We need a separate browserify bundling task for each destination bundle, so
+ * we pass it in to this function which then attaches the correct task to gulp
+ * under its name (for example, makeJSTask('index.js') would create a task
+ * 'index.js' that creates a destination bundle 'js/index.js' and starts
+ * watching for changes.
+ *
+ * @param {string} fname The filename of the entrypoint into the bundle,
+ * relative to `${srcDir}`/js/.
+ */
+var makeJSTask = function (fname) {
+  var handleErr = (err) => {
+    util.log(util.colors.red(`Browserify Error: ${err.message}`));
+  };
+
+  var customOpts = {
+    entries: [`${srcDir}/js/${ fname }`],
+    // set to true for sourcemapping
+    debug: false,
```

Comment:
This is mostly just sugar, but I keep a [fancy gulpfile around][1] that allows for two tasks, one that builds source maps and one that doesn't, with minimal code reuse.

[1]: https://gist.github.com/jez/edcc7d94028bf9a5a4fd